web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -50,54 +47,46 @@ export class LayerDialogComponent implements OnDestroy {
   layer?: Layer;
   layerName!: string;
   projectId?: string;
-  activeProject$: Observable<Project>;
   subscription: Subscription = new Subscription();
   fieldTypes = FieldType;
   fields: List<Field>;
 
   constructor(
-    @Inject(MAT_DIALOG_DATA) data: { layerId: string },
+    @Inject(MAT_DIALOG_DATA)
+    data: {
+      layerId: string;
+      projectId: string;
+      layers: Map<string, Layer>;
+      createLayer: boolean;
+    },
     private dialogRef: MatDialogRef<LayerDialogComponent>,
-    private projectService: ProjectService,
     private dataStoreService: DataStoreService,
     private router: Router,
     private confirmationDialog: MatDialog
   ) {
     this.lang = 'en';
     // Disable closing on clicks outside of dialog.
     dialogRef.disableClose = true;
+    this.fields = List<Field>();
     this.layerId = data.layerId;
-    this.activeProject$ = this.projectService.getActiveProject$();
-    this.subscription.add(
-      this.activeProject$.subscribe(project => {
-        this.onProjectLoaded(project);
-      })
-    );
-    const fields = List<Field>();
-    const fieldId = this.dataStoreService.generateId();
-    this.fields = fields.push(
-      new Field(
-        fieldId,
-        1,
-        StringMap({
-          en: '',
-        }),
-        false,
-        0,
-        undefined
-      )
-    );
+    if (data.layers) {
+      this.initProject(data.projectId, false, data.layers);
+    }
+    if (data.createLayer) {
+      this.initProject(data.projectId, true);
+    }
   }
 
   addQuestion() {
     const fieldId = this.dataStoreService.generateId();
     this.fields = this.fields.push(
       new Field(
         fieldId,
-        1,
+        FieldType.TEXT,
         StringMap({
           en: '',
         }),
+        /* required= */
```

Comment:
Thanks! What is the last arg in this call? If it's optional just leave it out instead of specifying `undefined`.